icu4j/main/classes/core/src/com/ibm/icu/impl/units/SingleUnitImpl.java

```diff
@@ -0,0 +1,177 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html#License
+/*
+ *******************************************************************************
+ * Copyright (C) 2004-2020, Google Inc, International Business Machines
+ * Corporation and others. All Rights Reserved.
+ *******************************************************************************
+ */
+
+package com.ibm.icu.impl.units;
+
+import com.ibm.icu.util.MeasureUnit;
+import jdk.nashorn.internal.runtime.regexp.joni.exception.InternalException;
+
+public class SingleUnitImpl {
+    /**
+     * Simple unit index, unique for every simple unit, -1 for the dimensionless
+     * unit. This is an index into a string list in unit.txt {ConversionUnits}.
+     * <p>
+     * The default value is -1, meaning the dimensionless unit:
+     * isDimensionless() will return true, until index is changed.
+     */
+    private int index = -1;
+    /**
+     * SimpleUnit is the simplest form of a Unit. For example, for "square-millimeter", the simple unit would be "meter"Ò
+     * <p>
+     * The default value is "", meaning the dimensionless unit:
+     * isDimensionless() will return true, until index is changed.
+     */
+    private String simpleUnit = "";
+    /**
+     * Determine the power of the `SingleUnit`. For example, for "square-meter", the dimensionality will be `2`.
+     * <p>
+     * NOTE:
+     * Default dimensionality is 1.
+     */
+    private int dimensionality = 1;
+    /**
+     * SI Prefix
+     */
+    private MeasureUnit.SIPrefix siPrefix = MeasureUnit.SIPrefix.ONE;
+
+    public SingleUnitImpl clone() {
+        SingleUnitImpl result = new SingleUnitImpl();
+        result.index = this.index;
+        result.dimensionality = this.dimensionality;
+        result.simpleUnit = this.simpleUnit;
+        result.siPrefix = this.siPrefix;
+
+        return result;
+    }
+
+    public MeasureUnit build() {
+        MeasureUnitImpl measureUnit = new MeasureUnitImpl(this);
+        return measureUnit.build();
+    }
+
+    /**
+     * Generates an neutral identifier string for a single unit which means we do not include the dimension signal.
+     *
+     * @throws InternalException if a dimensionless SingleUnitImpl
+     */
+    public String getNeutralIdentifier() {
```

Comment:
How about `getSimpleUnitIdentifier` or just `getSimpleIdentifier`? This is also in line with the terminology of the specification: https://www.unicode.org/reports/tr35/tr35-general.html#Unit_Identifiers - whereas "neutral" means nothing to me in this context.

Reply:
in this function we return the identifier without caring about the sign of the dimensionality. therefore, I called it neutral. we can call it absolute too.

Reply:
How is it different from a "simple unit identifier" in the spec at https://www.unicode.org/reports/tr35/tr35-general.html#Unit_Identifiers ?

Reply:
the identifier can be:
"square-millimeter" which is not simple unit

Reply:
It does return dimensionality (in the sense of "square-", "cubic-"), it skips just the sign (+/-), I had misunderstood the comment string.

Reply:
(This function maps to C++'s "serializeSingle".)

Reply:
Marking as resolved, I imagine the naming might get revisited as part of https://github.com/icu-units/icu/issues/87